lib/faraday/connection.rb

```diff
@@ -79,15 +79,22 @@ def initialize(url = nil, options = nil)
       @params.update(options.params)   if options.params
       @headers.update(options.headers) if options.headers
 
-      @proxy = @no_proxy = nil
+      @proxy = nil
       proxy(options.fetch(:proxy) {
-        uri = ENV['http_proxy']
-        if uri && !uri.empty?
-          uri = 'http://' + uri unless uri.downcase.start_with?('http')
-          uri
+        uri = nil
+        if URI.parse("").respond_to?(:find_proxy)
```

Comment:
According to the documentation, `find_proxy` is supported in Ruby 1.9.3 (https://ruby-doc.org/stdlib-1.9.3/libdoc/open-uri/rdoc/URI/Generic.html). Since we dropped the support for older ruby version, I think this method should always be available? Do you get any failing test if you remove this check?